client/src/unifyfs.c

```diff
@@ -858,15 +858,25 @@ int unifyfs_gfid_stat(int gfid, struct stat* buf)
     size_t filesize;
     ret = invoke_client_filesize_rpc(gfid, &filesize);
     if (ret != UNIFYFS_SUCCESS) {
-        return ret;
+        /*
+         * If we're here, then we were able to get the fattr, but not the size.
+         * That happens when we have a local file that has not been laminated
+         * yet.  In that case, return filesize = 0.
+         */
+        filesize = 0;
     }
 
+    /*
+     * set the file size
+     *
+     * TODO: Why doesn't invoke_client_metaget_rpc() fill in the filesize to
```

Comment:
The answer here is that currently it is still be expensive to look up the file size, because we are fetching *all* key/value pairs and iterating over the full set to compute the max write offset.  However, we may need to get the file meta frequently when we don't care about the file size, so they are separated into two different rpcs.

Once a file has been laminated, we could also record the file size in the metadata, so that if we see the laminated bit set, we also know we can trust the file size field and avoid calling the filesize_rpc.

Reply:
Ok, I'm my next push, I'll cache the size when the file is laminated.